### ISSUE TITLE
Support creating repository from template

### ISSUE THREAD

**Issue by mwarkentin** [terraform-providers/terraform-provider-github#328]
This is the flipside of #327 - it would be great to be able to launch new repositories from existing template repositories.

**Note**: This API is currently in preview, not sure if that disqualifies it from TF provider support or not.

### Affected Resource(s)

- `github_repository`

### Terraform Configuration Files
```hcl
resource "github_repository" "repo" {
  name                = "repo-name"
  template_repository = "template-owner/template-repository"
}
```

or

```hcl
resource "github_repository" "repo" {
  name                = "repo-name"
  template_owner      = "owner"
  template_repository = "repo"
}
```

### Expected Behavior
New repository should be created based on the provided template.

### References

- #327

**Comment by skeggse** [comment]
This is solved by #254, no?

**Comment by mwarkentin** [comment]
@skeggse Looks like it, thanks for pointing that out! Guess it's been merged but not released in the provider yet.. will keep my eye out for that!
